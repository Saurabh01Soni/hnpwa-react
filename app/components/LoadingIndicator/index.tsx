import * as React from 'react';

import './LoadingIndicator.scss';

export interface LoadingIndicatorProps {
  active?: boolean;
  style?: object;
}

const defaultProps = {
  active: false,
  style: {},
};

const LoadingIndicator: React.SFC<LoadingIndicatorProps> = ({
  active,
  style,
 }) => {
  if (!active) {
    return null;
  }

  return (
    <div className="LoadingIndicator" style={style}>
      <div className="LoadingIndicator__col" />
      <div className="LoadingIndicator__col" />
      <div className="LoadingIndicator__col" />
    </div>
  );
};

LoadingIndicator.defaultProps = defaultProps;

export default LoadingIndicator;
