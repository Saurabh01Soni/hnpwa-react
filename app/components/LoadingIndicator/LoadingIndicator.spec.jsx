import React from 'react';
import { shallow } from 'enzyme';

import LoadingIndicator from './';

describe('<LoadingIndicator />', () => {
  it('should return null when render default', () => {
    const wrapper = shallow(<LoadingIndicator />);
    expect(wrapper.type()).toBe(null);
  });

  it('should return DOM tree when props.active === true', () => {
    const wrapper = shallow(<LoadingIndicator active />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should pass style to root div when props.style exist', () => {
    const style = { position: 'absolute', top: 10, left: 10 };
    const wrapper = shallow(<LoadingIndicator active style={style} />);
    expect(wrapper.props().style).toEqual(style);
  });
});
