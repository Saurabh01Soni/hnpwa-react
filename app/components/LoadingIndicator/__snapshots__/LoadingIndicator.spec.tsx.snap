// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoadingIndicator /> should return DOM tree when props.active === true 1`] = `
<div
  className="LoadingIndicator"
  style={Object {}}
>
  <div
    className="LoadingIndicator__col"
  />
  <div
    className="LoadingIndicator__col"
  />
  <div
    className="LoadingIndicator__col"
  />
</div>
`;
